### PR TITLE
Update `recv_stream` to handle all `Stream`s instead of just `tokio::mpsc::unbounded_channel`

### PR DESCRIPTION
**Pull request by MingweiSamuel**
Breaking change

### CODE REVIEW

File: book/surface_data.md [hunk 1]
```diff
@@ -12,11 +12,11 @@ For example, here we iterate through a vector of `usize` items and push them dow
 The Hello, World example above uses this construct.
 
 ## `recv_stream`
-More commonly, a flow is handling external data that comes in asynchronously via Rust's _tokio_ library.
-For this we need to allocate tokio _channels_ that allow Rust code to send data into the Hydroflow inputs. 
+More commonly, a flow is handling external data that comes in asynchronously via the _Tokio_ library.
```
Comment (jhellerstein) [inline book/surface_data.md:15]: add a link to tokio docs?

File: hydroflow/tests/surface_codegen.rs [hunk 2]
```diff
@@ -23,13 +20,13 @@ use tokio::task::LocalSet;
 // TODO(mingwei): Documentation articles.
 // TODO(mingwei): Find a way to display join keys
 
-fn recv_into_vec<T>(recv: &mut UnboundedReceiver<T>) -> Vec<T> {
-    std::iter::from_fn(|| recv.try_recv().ok()).collect()
+fn recv_into_vec<T>(recv: &mut UnboundedReceiverStream<T>) -> Vec<T> {
+    std::iter::from_fn(|| recv.as_mut().try_recv().ok()).collect()
```
Comment (jhellerstein) [inline hydroflow/tests/surface_codegen.rs:24]: This is kind of a syntactic bummer. Would we always need to say `.as_mut`? If so can we hide this? If not, what is the common case and can we have longer syntax only for the uncommon case?

**Review by jhellerstein — APPROVED** [review]

**Review by shadaj — APPROVED** [review]
Seems reasonable!

File: hydroflow/tests/surface_codegen.rs (hunk 2 above)
Comment (MingweiSamuel) [reply hydroflow/tests/surface_codegen.rs:24]: Common case shouldn't have the user interacting directly with `UnboundedReceiver[Stream]` much at all

**Pull request merged**